Add ADScaledCoupledTimeDerivative Kernel

edbworth opened the pull request:
Add new kernel object to the MOOSE framework
 - .C and .h files
 - 1 basic exodiff test and 2 heavy mms verification tests
 - Documentation

closes #31354 

@cticenhour @csdechant

tophmatthews committed:
add field variable to ConservativeAdvection

tophmatthews committed:
cleanup tests

tophmatthews committed:
test field advection and add scalar. Closes #30545

tophmatthews committed:
update doco and coverage

NamjaeChoi committed:
Kokkos build scripts #30655

NamjaeChoi committed:
GPU base header and initialization #30655

NamjaeChoi committed:
GPU basic data classes #30655

NamjaeChoi committed:
GPU mesh data #30655

NamjaeChoi committed:
Add initialize parameter to MOOSE classes to avoid their initialization during copy #30655

NamjaeChoi committed:
GPU assembly class #30655

NamjaeChoi committed:
GPU system classes #30655

NamjaeChoi committed:
Initialize GPU assembly and systems #30655

NamjaeChoi committed:
GPU datum object to privatize thread-wise data #30655

NamjaeChoi committed:
GPU variable coupleable system #30655

NamjaeChoi committed:
GPU tests #30655

NamjaeChoi committed:
GPU block and boundary restrictables #30655

NamjaeChoi committed:
GPU kernels and boundary conditions #30655

NamjaeChoi committed:
Framework-level GPU kernels and boundary conditions #30655

NamjaeChoi committed:
Module-level GPU kernels and boundary conditions #30655

NamjaeChoi committed:
Add GPU kernels and boundary conditions to the system #30655

NamjaeChoi committed:
GPU material system #30655

NamjaeChoi committed:
Framework-level GPU materials #30655

NamjaeChoi committed:
Module-level GPU materials #30655

NamjaeChoi committed:
Add GPU materials to the system #30655

NamjaeChoi committed:
Establish GPU calculation flow #30655

NamjaeChoi committed:
Change GPU to Kokkos, add namespace, address some minor comments #30655

NamjaeChoi committed:
Check early the existence of Kokkos objects added by actions and initialize Kokkos mesh conditionally #30655

NamjaeChoi committed:
Use private parameter and 'currently executing actions' flag instead of initialize flag in MOOSE classes #30655

NamjaeChoi committed:
Only sort all objects for Kokkos material warehouse #30655

NamjaeChoi committed:
Enhance docstrings, reorganize accessors, move function bodies outside for base classes #30655

NamjaeChoi committed:
Enhance docstrings, reorganize accessors, move function bodies outside for kernel classes #30655

NamjaeChoi committed:
Enhance docstrings, reorganize accessors, move function bodies outside for material classes #30655

NamjaeChoi committed:
Use pass-by-value for performance #30655

NamjaeChoi committed:
Do not allow instantiating vector and matrix outside Kokkos scope #30655

NamjaeChoi committed:
Fix node mapping in multi-process parallelization #30655

NamjaeChoi committed:
Parallel memory pool object #3065

NamjaeChoi committed:
Remove the use of MAX_DOF in assembly #30655

NamjaeChoi committed:
Add configure option for Kokkos #30655

NamjaeChoi committed:
Back to caching elemental data #30655

NamjaeChoi committed:
Use unordered containers for faster mesh update #30655

NamjaeChoi committed:
Reduce local fixed array sizes #30655

NamjaeChoi committed:
Remove support for embedded Kokkos using Makefile due to deprecation #30655

NamjaeChoi committed:
Add install documentations #30655

NamjaeChoi committed:
Address comments #30655

lindsayad committed:
Add compile_commands support for Kokkos sources #30655

lindsayad committed:
Need to perform all device restores before mods on host #30655

NamjaeChoi committed:
Substitute -Werror with -Werror=all-warnings #30655

NamjaeChoi committed:
Make kernels work regardless of MAX_DOF #30655

NamjaeChoi committed:
Add device option to Kokkos tests #30655

NamjaeChoi committed:
Fix creating multiple matrices #30655

NamjaeChoi committed:
Add Kokkos in test harness #30655

NamjaeChoi committed:
Fix stateful restart and recover #30655

NamjaeChoi committed:
Support disabling GPU capabilities for test coverage and debugging purposes #30655

NamjaeChoi committed:
Make code Valgrind-clean #30655

NamjaeChoi committed:
Make code GCC- and clang-clean #30655

NamjaeChoi committed:
Change -imacros to -include for nvcc compatibility #30916

NamjaeChoi committed:
Use libMesh predefined integer types #30655

NamjaeChoi committed:
Change prefixes to Kokkos in tests #30655

NamjaeChoi committed:
Check nodal definedness in nodal kernels and BCs #30655

NamjaeChoi committed:
Fix Kokkos dependency treatment in makefile #30655

NamjaeChoi committed:
Attempt to build separate libraries #30655

NamjaeChoi committed:
Addressing comments #30655

NamjaeChoi committed:
Increase coverage #30655

NamjaeChoi committed:
Increase coverage 2 #30655

NamjaeChoi committed:
Address comments 2 #30655

NamjaeChoi committed:
Addressing comments 3 #30655

NamjaeChoi committed:
Make load/store map non-static #30655

NamjaeChoi committed:
Addressing comments 4 #30655

NamjaeChoi committed:
Fix example/tutorial build #30655

NamjaeChoi committed:
Addressing comments 5 #30655

NamjaeChoi committed:
Factor out common residual and Jacobian loop #30655

NamjaeChoi committed:
Addressing comments 6 #30655

NamjaeChoi committed:
Change prefixes to Kokkos in sources #30655

NamjaeChoi committed:
Addressing comments 7 #30655

NamjaeChoi committed:
Fix restep test #30655

NamjaeChoi committed:
Improve material property design #30655

NamjaeChoi committed:
Resolve material property entanglement #30655

NamjaeChoi committed:
Remove unnecessary inline keyword #30655

NamjaeChoi committed:
Change devices to compute_devices #30655

NamjaeChoi committed:
Add object documentations #30655

NamjaeChoi committed:
Add syntax documentations #30655

lindsayad committed:
Initial user input

lindsayad committed:
Add space in DynamicSolidMechanicsPhysics md header to match QuasiStatic

lindsayad committed:
Add static method for computing Newmark-Beta vel/accel in InertialForce

I'll leverage this in the FSI velocity continuity class

lindsayad committed:
Create ADPenaltyVelocityContinuityNewmarkBeta

Refs #31248

lindsayad committed:
Leverage new class in user's ALE input

lindsayad committed:
Create test

lindsayad committed:
Make component parameter required

NamjaeChoi committed:
Update old syntax #30655

lindsayad committed:
Hopefully make nonAD FSI test more robust in the test harness

loganharbour committed:
Add errorExit helper

loganharbour committed:
Add --require-capabilities filter

refs #31310

loganharbour committed:
Move build to a helper so that it can be unit tested

refs #31310

loganharbour committed:
Remove unused import

loganharbour committed:
Correct unit tests for SystemExit changes

loganharbour committed:
Simplify method and correct spelling

refs #31310

loganharbour committed:
Add cache for capabilities and add getJobWithName helper

loganharbour committed:
Add test for --require-capability

refs #31310

NamjaeChoi committed:
Improve map implementation #30655

NamjaeChoi committed:
Error on displacement and adaptivity #30655

kyriv-lab committed:
correct qprime aux kernels Refs #31377

kyriv-lab committed:
use heat flux instead of linear heat rate for the duct Refs #31377

kyriv-lab committed:
propagate new parameter name Refs #31377

kyriv-lab committed:
Delete not needed kernels Refs #31377

kyriv-lab committed:
Delete not needed docs of deletd kernels Refs #31377

cticenhour committed:
Make August 2025 news public, add September template

Refs #11447

kyriv-lab committed:
Address reviewer comments Refs #31377

MengnanLi91 committed:
Add contribution from Joshua Hanophy and mine

Co-authored-by: Joshua Hanophy

alexanderianblair committed:
Fix incorrect attribute relabelling when coil domain has the highest attribute ID in MFEMCutTransitionSubMesh

alexanderianblair committed:
Add Markdown documentation for new classes

alexanderianblair committed:
Update MFEMSumAux to support optional scaling factors on either variable

kyriv-lab committed:
rename post processor Refs #31377

kyriv-lab committed:
add pp that calculates net power from duct Refs #31377

kyriv-lab committed:
add pp that calculates net power using TH balance of inlet and outlet Refs #31377

lindsayad committed:
Address Max review

- Directly use dynamic kernels instead of using deprecated action
- Remove erroneous assert
- Fix mistake in InertialForce velocity param

kyriv-lab committed:
Regold Refs #31377

kyriv-lab committed:
add test for the duct_heat_flux pp and regold Refs #31377

alexanderianblair committed:
Add design and issue number for new MFEMCutTransitionSubMesh test. Refs #31404

alexanderianblair committed:
Add missing header to MFEMScalarBoundaryIC. Refs #31404

alexanderianblair committed:
Add missing header to MFEMCutTransitionSubMesh. Refs #31404

alexanderianblair committed:
Add example demonstrating a strongly imposed global current constraint using a h-phi formulation. Refs #31404

alexanderianblair committed:
Remove superfluous comments in MFEMCutTransitionSubMesh. Refs #31404

lindsayad committed:
Add blurb about field split and static condensation

nmnobre committed:
Add summary of MFEM backend improvements

NamjaeChoi committed:
Remove subdomain local element ID map #30655

NamjaeChoi committed:
Cache reference element data separately #30655

GiudGiud committed:
Dont forget domainUOs for consumption of material properties
 refs #31418

GiudGiud committed:
const a few variables
Add TODOs for where to optimize
 refs #31418

GiudGiud committed:
Request matprop computations less often from the system base:
only if the BCs actually requested material properties
 refs #31418

GiudGiud committed:
Reinit with the same criterion as stateful property init
 refs #31418

alexanderianblair committed:
Restrict output of hphi_magnetostatic test to vacuum region. Refs #31404

alexanderianblair committed:
Remove unused variables and FESpaces in MFEMVacuumCutTransitionSubMesh test. Refs #31404

NamjaeChoi committed:
Use type aliases for Kokkos IDs #30655

GiudGiud committed:
Do not forget constraints needing boundary materials
refs #31418

GiudGiud committed:
Add guillaume and others' contributions for aug

alexanderianblair committed:
Enable the user to optionally label exterior boundary of submeshes created with MFEMDomainSubmesh. Refs #31404

alexanderianblair committed:
Add an MFEM magnetodynamic example to demonstrate usage of a boundary condition using a DomainSubMesh boundary. Refs #31404

GiudGiud committed:
InterfaceUO can use boundary properties

alexanderianblair committed:
Disable unsupported restep test for MFEMDomainSubMeshBoundary

GiudGiud committed:
Count constant properties as calling getMatProp
Adjust test description for how requests of properties now work

NamjaeChoi committed:
Increase coverage 3 #30655

NamjaeChoi committed:
Update module objects to match up the original versions #30655

NamjaeChoi committed:
Add module tests #30655

NamjaeChoi committed:
Fix SQA and confilicts #30655

GiudGiud committed:
Constraints are not threaded

kyriv-lab committed:
address reviewer comments Refs #31377

GiudGiud committed:
Use a dedicated routine on internal face on a boundary

GiudGiud committed:
Use matprop dependencies in getMatPropCalled, because IPHDG and their helper

GiudGiud committed:
Adapt test to less matprop unused computing

GiudGiud committed:
Compute material properties on boundaries for interface kernels

GiudGiud committed:
Switch to volumetric umat evaluation since test does use face properties

GiudGiud committed:
Turns out HDGKernels dont want materials on internal sides

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3181749/), step Generate coverage on 5c83c9c wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>1f9d31</th>
      <th colspan=4>#31425 5c83c9</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.66%</td>
      <td>85.66%</td>
      <td>+0.00%</td>
      <td>90.91%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>118023</td>
      <td>118033</td>
      <td>+10</td>
      <td>10</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>19760</td>
      <td>19761</td>
      <td>+1</td>
      <td>1</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/31425/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/31425/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/31425/coverage/xfem)

</details>



This comment will be updated on new commits.

alexanderianblair committed:
Apply suggestions from #31405 code review

Co-authored-by: Alex Lindsay <alexander.lindsay@inl.gov>

alexanderianblair committed:
Correct comments and unique_ptr construction in MFEMCutTransitionSubMesh. Refs #31404

alexanderianblair committed:
Use HYPRE_BigInt consistently for global vertex IDs in MFEMCutTransitionSubMesh. Refs #31404

hugary1995 committed:
Add ReactionForce Aux

Deprecate the `scaled` parameter in TagAuxBase. The deprecated parameter is replaced by `remove_variable_scaling` in TagVectorAux.

In solid mechanics, a new AuxKernel named `ReactionForceAux` is added where `remove_variable_scaling` is always set to true.

`ReactionForceAux.md` discusses the example usage to get reaction force.

ref #20482
ref #31357

NamjaeChoi committed:
Sanity check for Kokkos array initialization status #30655

GiudGiud commented on framework/doc/content/source/kernels/ADMatCoupledTimeDerivative.md:
```diff
@@ -0,0 +1,25 @@
+# ADMatCoupledTimeDerivative
```
"MatCoupled" isnt as clear as "CoefCoupled" or "ScaledCoupled" unforuntely. 

Wthiout looking I would have assumed it contributes dMatProp/dt to the equation for 'variable'

GiudGiud reviewed (changes requested):
I'd move that to the app that needs it. 
There's 20 or so possible declinations of this same kernel doing a time derivative. We cant host them all in the framework. If anything we have too many already

hugary1995 committed:
Apply suggestions from Ben

GiudGiud committed:
Add capability to convert only a few nodesets to sidesets
refs #31452

GiudGiud committed:
Add test for selective sideset addition
refs #31452

GiudGiud committed:
Allow this use of the ParsedNodesetGeneration for this case

GiudGiud committed:
Rework DG/HDG execution logic to limit the calls to expensive shouldComputeOnInternalSide
refs #31459

edbworth commented on framework/doc/content/source/kernels/ADMatCoupledTimeDerivative.md:
```diff
@@ -0,0 +1,25 @@
+The syntax is simple, taking its type
+(`ADMatCoupledTimeDerivative`), the kernel variable which the
+`ADMatCoupledTimeDerivative` residual is assigned to, the coupled variable `v`
+that the time derivative operator acts upon, and the material property 'mat_prop'. Example syntax can be found in the
```
When Making this change and building the docuemntation locally, I recieve many errors all of the form:

MooseDocs.base.readers (MainProcess): 
TOKENIZE ERROR: '/Reporters/StatisticsReporter' syntax was not recognized.
/Users/butted/projects/moose/modules/stochastic_tools/doc/stm_workshop/modules/stochastic_tools/intro_slides/index.md:47
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
47│[!param](/Reporters/StatisticsReporter/ci_method)                                               │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


Any thoughts on how to fix this?

GiudGiud replied:
that looks unrelated

there are many examples of using `[!param]` through the docs btw

hugary1995 committed:
Support partial homogenization constraint in SolidMechanics physics

close #31462

magnoxemo committed:
adding element hierarchy aux kernel
closes #31469

magnoxemo committed:
adding the tests
refs #31469

hugary1995 committed:
Address comments from Alex

hugary1995 committed:
Remove error check in ReactionForceAux that's no longer necessary

hugary1995 committed:
Fix test spec

GiudGiud commented on framework/doc/content/source/kernels/ADScaledCoupledTimeDerivative.md:
```diff
@@ -0,0 +1,22 @@
+# ADScaledCoupledTimeDerivative
+
+The `ADScaledCoupledTimeDerivative` kernel is an extension of the
+[`ADCoupledTimeDerivative`](/ADCoupledTimeDerivative.md) kernel that consumes a material property.
```
```suggestion
[`ADCoupledTimeDerivative`](/ADCoupledTimeDerivative.md) kernel that has a scaling factor.
This factor can be:

- a material property using the [!param](/Kernels/ADScaledCoupledTimeDerivative/mat_prop) parameter
```

GiudGiud commented on framework/include/kernels/ADScaledCoupledTimeDerivative.h:
```diff
@@ -0,0 +1,28 @@
+#include "ADCoupledTimeDerivative.h"
+
+/**
+ * This calculates the time derivative of a coupled variable scaled by material properties.
```
```suggestion
 * This calculates the time derivative of a coupled variable scaled by a material property.
```

GiudGiud commented on framework/src/kernels/ADScaledCoupledTimeDerivative.C:
```diff
@@ -0,0 +1,35 @@
+ADScaledCoupledTimeDerivative::validParams()
+{
+  auto params = ADCoupledTimeDerivative::validParams();
+  params.addClassDescription("Extension of the ADCoupledTimeDerivative kernel that consumes an "
+                             "arbitrary material property");
```
update

GiudGiud commented on framework/src/kernels/ADScaledCoupledTimeDerivative.C:
```diff
@@ -0,0 +1,35 @@
+  params.addClassDescription("Extension of the ADCoupledTimeDerivative kernel that consumes an "
+                             "arbitrary material property");
+
+  params.addRequiredParam<MaterialPropertyName>("mat_prop", "Name of the material property");
```
```suggestion
  params.addRequiredParam<MaterialPropertyName>("mat_prop", "Name of the material property factor");
```

GiudGiud reviewed (commented):
can you also tone down (a lot) the MMS tests (I d probably remove them tbh)? This is 30s of extra testing for something that has close to 0 chances to ever be wrong

magnoxemo committed:
adding better test requirement description, removing empty whitespace, fix grammar
refs #31469

edbworth commented:
> can you also tone down (a lot) the MMS tests (I d probably remove them tbh)? This is 30s of extra testing for something that has close to 0 chances to ever be wrong

These tests are designated as heavy, but they take too long still? Any ideas on modifying these to be less computationally intensive or do you agree that we should remove them? @csdechant

NamjaeChoi committed:
Define special passkey-protected constructors for functor copy #30655

GiudGiud committed:
Add a test material that will error on boudnaries on which we dont want to execute

GiudGiud committed:
Add tests for not computing materials where we dont need them
- face materials on boundaries not using matprops
- face materials on interfaces not using matprops

NamjaeChoi committed:
Addressing comments 8 #30655

magnoxemo committed:
rename the auxkernel, add options for p adaptivity
refs #31469

loganharbour committed:
Resolve mooseError issues introduced in #31396

jessecarterMOOSE committed:
use LDFLAGS when compiling capabilities target, closes #31481

maythuwin1313 committed:
Update libmesh submodule

refs #31560

Summary of changes:

- Limited support for untangling of tangled meshes in
  `VariationalMeshSmoother`
- Support for smoothing meshes including pyramids with
  `VariationalMeshSmoother`
- Added `SparseMatrix::read_coreform_hdf5()` method for reading CSR
  sparse matrix data from an HDF5-based format
- Default to syncing C++ streams for more reliable console output in
  cases like threaded output
- Made `libMesh::out` and `libMesh::err` thread-safe even when
  emitting non-ASCII (unicode, color codes) output
- Made `prepare_for_use()` optional after mesh stitching.  This will
  enable much faster parallel generation of complex meshes in certain
  MOOSE workflows.
- Added optional parameter to `build_side_list_from_node_list()` to
  allow only some nodesets to be converted rather than all
- Added `Parameters::Value::type_info()` method
- Updated autotools to their latest versions.  This fixes a linker bug
  that was preventing libMesh compilation with some newer software
  stacks.
- Use `MatGetValue` instead of `MatGetRow` in
  `PetscMatrix::operator()`, to improve compatibility and efficiency
- `PerfLog` performance logging of Netgen tetrahedralization
- Bug fixes for parallel `StaticCondensation` operation
- Bug fix for certain use cases of `PetscVector` constructed directly
  from PETSc `Vec`
- Bug fix for triangulation of boundaries with quadratic edges
- Bug fix for an adaptive refinement failure case on distributed meshes
- Fixed handling of linearly-dependent data in reduced basis EIM
  training
- Fixed small `PetscViewer` memory leaks in `print_matlab()` methods

kyriv-lab committed:
define direction of gravity Refs #31483

lindsayad committed:
Restore passkey for device construction #30655

lindsayad committed:
Use passkey for isKokkosObject #30655

loganharbour committed:
Disable debuginfod with valgrind tests

refs #31493

tophmatthews committed:
Rename files

kyriv-lab committed:
add test Refs #31483

tophmatthews committed:
adify conservativeadvectionbc

GiudGiud committed:
Add a check on variable type
Fix up input
Add missing issue

tophmatthews committed:
add tests

tophmatthews committed:
doco. closes #31503

tophmatthews committed:
fix templating

tophmatthews committed:
doco fixes

lynnmunday committed:
adding cutter mesh output to 3D xfem cutter object.  This commit creates a base class that will have some functionality needed by the 2D and 3D mesh cutter classes.

lynnmunday committed:
adding mesh cutter output to 3D mesh cutter user object with test.  closes #31507

tophmatthews committed:
make negSpeedQp virtual. Closes #31508

GiudGiud committed:
Address Alex review

- better comments
- rely on getMatpropCalled
- fixup the name
- rename input

Co-authored-by: Alex Lindsay <alexander.lindsay@inl.gov>

jessecarterMOOSE committed:
fix logic in building time sequence, closes #31511

GiudGiud committed:
Add missing setup calls for exotic objects in NLSystem

Avoid allocating threaded warehouses for non-threaded systems
refs #31514

lynnmunday committed:
trying to shift nodes in initial cutter mesh to make cutter mesh connectivity deterministic so exodiff will pass

GiudGiud committed:
Convert the GenericConstantVectorMaterial to use std::vector<Real> properties
RealVectorValue is handled by GenericConstantRealVectorValue

GiudGiud committed:
Add a basic test for the std::vector<Real> material properties

GiudGiud committed:
Add 1st component output for std::vector<Real> properties
refs #31519

GiudGiud committed:
Add support for outputting all components of a vector Real matprop
refs #31519

GiudGiud committed:
Better and safer material output behavior

GiudGiud committed:
Avoid hard-deprecating the GenericConstantVectorMaterial
Use the StdVector particulate instead

maythuwin1313 committed:
Add intel compiler into mpi container

refs #31560

nmnobre committed:
Accommodate changes in PETSc 3.24.0 (#31534)

nmnobre committed:
Fix MFEM assertion (#31405)

lynnmunday committed:
adding the ability to read or create the cutter mesh from the mesh block

loganharbour committed:
Update code server version

refs #31535

loganharbour committed:
Update moose language support plugin

refs #31535

loganharbour committed:
Allow openmpi to run as root in code-server

refs #31535

loganharbour committed:
Bump moose-dev version

- Update code-server in moose-dev apptainer
- Update moose-language-support in code-server moose-dev apptainer
- Allow openmpi as root in code-server in moose-dev-apptainer

refs #31535

loganharbour committed:
Update versioner hashes

nmnobre committed:
Swap MFEM_ASSERT by mooseAssert (#31534)

NamjaeChoi committed:
Improve Kokkos initialization settings #30655

lynnmunday committed:
converting 2D mesh cutter input files to use a mesh generator to create the mesh cutter

NamjaeChoi committed:
Mark some MFEM tests as valgrind heavy #31539

GiudGiud committed:
Address Alex review

- more const
- more auto
- keep const ref of property names
- size_t for size ints

Also:
- add link to std::vector matprop mat in both realvectorvalue matprop mat
- correct templating for matoutputaction

Co-authored-by: Alex Lindsay <alexander.lindsay@inl.gov>

tophmatthews committed:
review comments

NamjaeChoi committed:
Add Kokkos-MOOSE in September newsletter #30655

NamjaeChoi committed:
Add links to object syntax pages in the Kokkos main syntax page #30655

loganharbour committed:
Resolve issues with require capability cli arg

refs #31310

loganharbour committed:
Use subprocess run instead to avoid hanging thread

lynnmunday committed:
updating documentation fo rusing mesh generator cutter mesh

maythuwin1313 committed:
Update wasp submodule

refs #31560

tophmatthews committed:
review comments

lynnmunday committed:
fixing test for newly deprecated meshcutter parameter

lindsayad committed:
Put libmesh libraries after capabilities sources

Refs #31481 and #31523

loganharbour committed:
Rename from --require-capability to --only-tests-that-require

refs #31310

loganharbour committed:
Add newsletter entry for --only-tests-that-require

refs #31310

tophmatthews committed:
error if member variable is used as velocity

tophmatthews committed:
fix doco

maythuwin1313 committed:
Remove whitespace in cpp-peglib

This errors with Intel oneAPI

refs #31560

maythuwin1313 committed:
Remove template from constructor and destructor

This errors with gcc14

refs #31560

maythuwin1313 committed:
Add Intel compiler capability

refs #31560

maythuwin1313 committed:
Set correct openmp flag for Intel compiler

refs #31560

tophmatthews committed:
add petsc jacobian tests

maythuwin1313 committed:
Add PythonFMU in moose-dev.def

refs #31560

loganharbour committed:
Remove libmesh dependency for capabilities library and fix test

refs #31394, #31481, #31523, #31546

loganharbour committed:
Try to use the underlying compiler instead of the mpi wrapper

refs #31394, #31481, #31523, #31546

maythuwin1313 committed:
Bump versions of mpi, libmesh-vtk, petsc, libmesh, wasp and moose-dev

- Add intel compiler to mpi container
- Update libmesh submodule
- Update wasp submodule
- Add PythonFMU to moose-dev container

refs #31560

maythuwin1313 committed:
Update versioner hashes

refs #31560

NamjaeChoi committed:
Add framework unity build rule for Kokkos #30655

tophmatthews committed:
small doco typo

roystgnr committed:
Import numpy.typing as npt

This fixes #31564 for me

GiudGiud committed:
Use a stronger preconditioner for more reproducible results at high MPI count
refs #11852 #30545

lynnmunday committed:
Apply suggestions from code review
Correct param syntax.

Co-authored-by: Ben Spencer <benjamin.spencer@inl.gov>

lynnmunday committed:
combined documentation for EqualValueEmbeddedConstraint and made shouldApply final closes #31576

bwspenc committed:
Fix xfem tests that were diffing on Mac closes #31578

Both were diffing with relative tolerances in the 1e-5 range.
Used better solver settings to get them to converge much tighter and
regolded.

tophmatthews committed:
correct jacobian

GiudGiud reviewed (changes requested):
while I really like that you did all this MMS testing to make sure everything was correct, we don't need this much for a simple kernel. I think the standard for rolling out MMS at this time is a full equation (like NS, the various EM formulations, etc)

edbworth committed:
Add new ADScaledCoupledTimeDerivative kernel to the MOOSE framework

.C and .h files
1 basic exodiff test with gold file
Documentation and markdown file
Removed MMS testing
Incorporated suggested changes from Guillaume
closes #31425

edbworth committed:
Tring to fix merge conflicts

edbworth commented:
Closed this due to confusion with branch name vs kernel name. Will open new pull request shortly.
